sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -88,6 +88,7 @@ abstract class Optimizer(catalogManager: CatalogManager)
         LimitPushDownThroughWindow,
         ColumnPruning,
         GenerateOptimization,
+        PushPartialAggregationThroughJoin,
```

Comment:
should we do this post Collpase project rule ?
As we could miss this case : 

Agg -> Proj -> Proj -> Join (inner / non-broadcastable)

Reply:
It should work in the next iteration.